ludwig/explain/base.py

```diff
@@ -0,0 +1,74 @@
+from abc import ABCMeta, abstractmethod
+from typing import List, Tuple
+
+import pandas as pd
+
+from ludwig.api import LudwigModel
+from ludwig.constants import BINARY, CATEGORY, TYPE
+from ludwig.explain.util import Explanation, prepare_data
+from ludwig.utils.torch_utils import get_torch_device
+
+DEVICE = get_torch_device()
+
+
+class Explainer(metaclass=ABCMeta):
+    def __init__(self, model: LudwigModel, inputs_df: pd.DataFrame, sample_df: pd.DataFrame, target: str):
+        """Constructor for the explainer.
+
+        # Inputs
+
+        :param model: (LudwigModel) The LudwigModel to explain.
+        :param inputs_df: (pd.DataFrame) The input data to explain.
+        :param sample_df: (pd.DataFrame) A sample of the ground truth data.
+        :param target: (str) The name of the target to explain.
+        """
+        model.model.to(DEVICE)
```

Comment:
Perhaps this should be a check that `model.model.device == DEVICE`?   The reason I say that is because I would be surprised to see that constructing Explainer(model, ...) would in-place modify model.
If we need to keep `model.to(DEVICE)` here, mention it in the doc string.

Reply:
Moved device placement from base class to `IntegratedGradientsExplainer.explain`, as it is not used by `GBMExplainer`.